src/wp-includes/option.php

```diff
@@ -364,6 +364,157 @@ function get_options( $options ) {
+ * @param string|bool $autoload Autoload value to control whether to load the option when WordPress starts up.
+ *                              Accepts 'yes'|true to enable or 'no'|false to disable.
+ * @return bool True if the autoload value was modified, false otherwise.
+ */
+function wp_set_option_autoload( $option, $autoload ) {
+	global $wpdb;
+
+	$autoload = ( 'no' === $autoload || false === $autoload ) ? 'no' : 'yes';
+
+	/*
+	 * Determine the current autoload value for the option.
+	 * If the value is empty, it means the option does not exist in the database.
+	 * If the old and new value are the same, no need to update.
+	 */
+	$old_autoload = $wpdb->get_var( $wpdb->prepare( "SELECT autoload FROM $wpdb->options WHERE option_name = %s", $option ) );
+	if ( ! $old_autoload || $old_autoload === $autoload ) {
+		return false;
+	}
+
+	// Update the autoload value.
+	$result = $wpdb->update(
+		$wpdb->options,
+		array( 'autoload' => $autoload ),
+		array( 'option_name' => $option )
+	);
+	if ( ! $result ) {
+		return false;
+	}
+
+	// Update caches as necessary based on the autoload change.
+	$alloptions = wp_load_alloptions( true );
+	if ( 'no' === $autoload ) {
+		// If set in alloptions, remove and migrate to individual cache.
+		if ( isset( $alloptions[ $option ] ) ) {
+			wp_cache_set( $option, $alloptions[ $option ], 'options' );
+			unset( $alloptions[ $option ] );
+			wp_cache_set( 'alloptions', $alloptions, 'options' );
+		}
+	} else {
+		// If set in cache, migrate to alloptions and delete individual cache.
+		$value = wp_cache_get( $option, 'options' );
+		if ( false !== $value ) {
+			$alloptions[ $option ] = $value;
+			wp_cache_delete( $option, 'options' );
+			wp_cache_set( 'alloptions', $alloptions, 'options' );
+		}
+	}
+
+	return true;
+}
+
+/**
+ * Sets the autoload value for multiple options in the database.
+ *
+ * Autoloading too many options can lead to performance problems, especially if the options are not frequently used.
+ * This function allows modifying the autoload value for multiple options without changing the actual option value.
+ * This is for example recommended for plugin activation and deactivation hooks, to ensure any options exclusively used
+ * by the plugin which are generally autoloaded can be set to not autoload when the plugin is inactive.
+ *
+ * @since 6.4.0
+ * @see wp_set_option_autoload()
+ *
+ * @global wpdb $wpdb WordPress database abstraction object.
+ *
+ * @param array       $options  List of option names. Expected to not be SQL-escaped.
+ * @param string|bool $autoload Autoload value to control whether to load the options when WordPress starts up.
+ *                              Accepts 'yes'|true to enable or 'no'|false to disable.
+ * @return array Associative array of all provided $options as keys and boolean values for whether their autoload value
+ *               was updated.
+ */
+function wp_set_options_autoload( $options, $autoload ) {
```

Comment:
This signature only allows you to bulk set to 'no' or 'yes'. What about `[ 'foo' => 'yes', 'bar' => 'no' ]`?

Reply:
@boonebgorges In principle, I like this idea for flexibility. That said, I think the main use-case for bulk-modifying options' autoload values is to e.g. set all of a plugin's options to `no` when it's being deactivated, or the respective options to `yes` when it's reactivated. So I'd prefer to at least maintain the ability to provide multiple options, but a single value for all of them.

To enable the use-case you're outlining, what do you think about the following?
* Rename/change this function to `wp_set_option_autoload_values( $options )`, requiring the associative array you're outlining.
* Make `wp_set_options_autoload( $options, $autoload )` a wrapper for that function which just assmbles the associative array, setting all values to the single autoload value.
* Per @joemcgill's feedback below, `wp_set_option_autoload( $option, $autoload )` would already be a wrapper for the single main function.

Reply:
Yes, these changes seem reasonable to me. Thanks for collating the feedback :)

Reply:
@boonebgorges Implemented the new function and more tests (due to the added flexibility) in https://github.com/WordPress/wordpress-develop/pull/5069/commits/b25ead5420dd0099f04b97d2789f9c03b2f10064.